test/e2e/suite/certificates/othernamesan.go

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright 2020 The cert-manager Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package certificates
+
+import (
+	"context"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/asn1"
+	"encoding/pem"
+	"time"
+
+	"github.com/cert-manager/cert-manager/e2e-tests/framework"
+	e2eutil "github.com/cert-manager/cert-manager/e2e-tests/util"
+	cmapi "github.com/cert-manager/cert-manager/pkg/apis/certmanager/v1"
+	cmmeta "github.com/cert-manager/cert-manager/pkg/apis/meta/v1"
+	"github.com/cert-manager/cert-manager/test/unit/gen"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = framework.CertManagerDescribe("othername san processing", func() {
```

Comment:
suggestion: I think e2e tests like this are the most valuable type, and I like that this test exists.

I think we have a (not well documented) suite of e2e tests which will try testing a feature against inbuilt issuers which support it. See https://github.com/cert-manager/cert-manager/blob/8da699a73593206ce40def2b2f1e38494e5748cb/test/e2e/suite/conformance/certificates/tests.go

If we moved this test there, we could test this against venafi, vault, CA issuers and selfisigned. If it doesn't work for one of those, it can be disabled AFAIK.

Is it worth moving this over there?